tests/integration/test_charm.py

```diff
@@ -30,8 +31,15 @@ async def test_build_and_deploy(ops_test: OpsTest):
     await ops_test.model.deploy(
         charm_under_test, resources=resources, trust=True, config=OIDC_CONFIG
     )
+
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME], status="blocked", raise_on_blocked=False, timeout=60 * 10
```

Comment:
In general I prefer not to check the unit status inside integration tests as the status can be affected by different events and configurations, as well as block the entire integration execution because of that, plus you are already testing this in the unit tests.

Reply:
Regarding this, I get it but how should we handle when a charm is expected to go to Blocked? Writing this though felt weird tbh and maybe we should never expect this. Should we (for integration tests) deploy the charm then preconfigured with public_url? I am OK with this. 

I think though bundle tests should use `Blocked` (instead of preconfiguring the charm) in order to verify what expected user is expected to go through. That's the reason [behind this](https://github.com/canonical/bundle-kubeflow/blob/f94f67d42dc403f38287237d0f9c7631c8ba0de4/tests/integration/test_bundle_deployment.py#L32-L38) and that's what I was thinking of adding to [those bundle tests](https://github.com/canonical/bundle-kubeflow/blob/main/tests-bundle/1.7/test_release_1-7.py#L59-L65) as well (after this PR is merged)

Reply:
I am in favour of deploying with the configuration option as that would setup the charm for working correctly in the testing environment, and let the unit status checks in the unit tests. You'd be testing that the unit goes to a desired status under a very controlled environment, whereas in integration we'd have multiple cases for a unit to go blocked.

Reply:
Pushed a commit for this then